bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -539,13 +552,32 @@ func (r *KubeadmConfigReconciler) joinWorker(ctx context.Context, scope *Scope)
 			UseExperimentalRetry: scope.Config.Spec.UseExperimentalRetryJoin,
 		},
 		JoinConfiguration: joinData,
-	})
+	}
+
+	var bootstrapJoinData []byte
+	switch scope.Config.Spec.Format {
+	case bootstrapv1.Ignition:
+		ign, _, err := ignition.NewNode(&ignition.NodeInput{
+			NodeInput: nodeInput,
+			Ignition:  scope.Config.Spec.Ignition,
+		})
+
+		if err != nil {
+			scope.Error(err, "Failed to generate Ignition user data for the worker node")
+			return ctrl.Result{}, err
+		}
+
+		bootstrapJoinData = ign
```

Comment:
```suggestion
		bootstrapJoinData, _, err = ignition.NewNode(&ignition.NodeInput{
			NodeInput: nodeInput,
			Ignition:  scope.Config.Spec.Ignition,
		})
```
nit, I think we don't have to handle ignition differently then cloud init here

Reply:
Fixed and added a test to ensure correct handling.

Reply:
I'll mark as resolved.